src/views/ApartmentApp/components/StaffMenu/index.js

```diff
@@ -0,0 +1,174 @@
+ * @param {*} props.authentication The user authentication
+ * @param {StudentProfileInfo} props.userProfile The student profile info of the current user
+ * @returns {JSX.Element} JSX Element for the staff menu web page
+ */
+const StaffMenu = ({ userProfile, authentication }) => {
+  const [loading, setLoading] = useState(true);
+
+  /** @type {[ApplicationDetails[], React.Dispatch<React.SetStateAction<ApplicationDetails[]>>]} ApplicationDetails */
+  const [applications, setApplications] = useState([]);
+
+  /** @type {[ApplicationDetails[], React.Dispatch<React.SetStateAction<ApplicationDetails[]>>]} Array of application details, after being formatted for use with react-csv */
+  const [applicationJsonArray, setApplicationJsonArray] = useState([]);
+
+  /** @type {[ApartmentApplicant[], React.Dispatch<React.SetStateAction<ApartmentApplicant[]>>]} Array of applicant info, after being formatted for use with react-csv */
+  const [applicantJsonArray, setApplicantJsonArray] = useState([]);
+
+  /** @type {[ApartmentChoice[], React.Dispatch<React.SetStateAction<ApartmentChoice[]>>]} Array of apartment choice info, after being formatted for use with react-csv */
+  const [apartmentChoiceJsonArray, setApartmentChoiceJsonArray] = useState([]);
+
+  const [dateStr, setDateStr] = useState('');
+  const filePrefix = 'apartapp';
+
+  /**
+   * Attempt to load all existing applications for the current semester
+   */
+  const loadAllCurrentApplications = useCallback(async () => {
+    setLoading(true);
+    let applicationDetailsArray = await housing.getAllApartmentApplications();
+    setApplications(applicationDetailsArray ?? []);
+    setLoading(false);
+  }, []);
+
+  useEffect(() => {
+    loadAllCurrentApplications();
+
+    // Generate string of today's date in ISO format for use in CSV filename
+    let date = new Date();
+    let dateStr = DateTime.fromJSDate(date).toISODate();
+    setDateStr(dateStr);
+  }, [userProfile, loadAllCurrentApplications]);
+
+  /**
+   * Generate arrays of objects to be converted to a CSV
+   * @param {ApplicationDetails[]} applicationDetailsArray an array of ApplicationDetails objects
+   */
+  const generateCSVData = useCallback((applicationDetailsArray) => {
+    let applicationsForCsv = [];
+    let applicantsForCsv = [];
+    let apartmentChoicesForCsv = [];
+    applicationDetailsArray.forEach((applicationDetails) => {
+      // Only add the applications that have been submitted
+      if (applicationDetails.DateSubmitted) {
+        let { Applicants, ApartmentChoices, ...filteredApplicationDetails } = applicationDetails;
+        applicationsForCsv.push(filteredApplicationDetails);
+
+        Applicants.forEach((applicant) => {
+          if (applicant.ApplicationID) {
+            applicantsForCsv.push(applicant);
+          } else {
+            applicantsForCsv.push({
+              ApplicationID: applicationDetails.ApplicationID,
+              ...applicant,
+            });
+          }
+        });
```

Comment:
You can follow the code for appending apartment choices to remove the `if` block. 
```suggestion
        Applicants.forEach((applicant) =>
          applicantsForCsv.push({
            ApplicationID: applicant.ApplicationID ?? applicationDetails.ApplicationID,
            ...applicant,
          }),
        );
```

Additionally, because the spread operator will overwrite earlier properties only if the property exists on the object being spread, you could remove the explicit set of `applicant.ApplicationID`. However, I think this somewhat obfuscates the expected path of the code, so I left it the more obvious way above.

Reply:
Also, I forgot to mention this, but the CSV output for Applicants has the Profile as simply `[object Object]`, because the value for that column is an object.
![image](https://user-images.githubusercontent.com/35319956/113918164-178bac80-97b0-11eb-9037-3769fd82fced.png)

Reply:
Correct, that is why I filter out the applicant objects from the Applications CSV array, so we should do the same with the Profile objects here

Reply:
That overwriting behavior of the spread operator is actually preferrable in this case, since we only want to set the ID if already null or undefined

Reply:
RE: Profile filtering - that should be easy, I'll implement and commit it.

RE: setting the ApplicationID - I suppose applicant.ApplicationID should never be different from applicationDetails.ApplicationID, right? Is it expected to have a case where the applicant object has a null/undefined ApplicationID?

Reply:
I did not actually know that the order of where you put the spread operator changes it's behavior like that, so that is very good to know. I should go through all the places I have used and make sure to take that into account

Reply:
It also looks like I should be filtering out the StudentID as well. I will add that to the branch I am currently working in